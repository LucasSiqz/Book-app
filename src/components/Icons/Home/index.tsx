const Home = () => (
  <svg
    width="18"
    height="19"
    viewBox="0 0 18 19"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M1 6.22222L9 0L17 6.22222V16C17 16.9818 16.2041 17.7778 15.2222 17.7778H2.77778C1.79594 17.7778 1 16.9818 1 16V6.22222Z"
      stroke="black"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
    <path
      d="M6.33333 17.7778V8.88892H11.6667V17.7778"
      stroke="black"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
)

export default Home
