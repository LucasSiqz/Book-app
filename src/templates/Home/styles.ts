import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
`

export const HeaderSection = styled.section`
  padding: 5rem 2rem 3rem 2rem;

  > div:first-child {
    margin-bottom: 3rem;
  }
`

export const DiscoverSection = styled.section`
  position: relative;
  padding-left: 2rem;
  margin-bottom: 3rem;

  > div:first-child {
    padding-right: 2rem;
  }
`

export const CurrentlyReadingSection = styled.section`
  > div:first-child {
    padding: 0 2rem;
  }

  > div:nth-child(2) {
    width: 89%;

    @media (min-width: 750px) {
      margin: 5rem auto 7rem auto;
      transform: scale(1.5);
      width: 40%;
    }

    @media (min-width: 1000px) {
      width: 35%;
    }
  }
`

export const ReviewsSection = styled.section`
  padding-left: 2rem;
  padding-right: 2rem;
  padding-bottom: 6rem;
`

export const Grid = styled.section`
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(9rem, 1fr));
  grid-column-gap: 2rem;
  grid-row-gap: 4rem;
  padding: 0 2rem 9rem 2rem;
`
