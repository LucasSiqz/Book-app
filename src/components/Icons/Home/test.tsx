import { render } from 'utils/test-utils'

import Home from '.'

describe('<HomeIcon />', () => {
  it('should render correctly', () => {
    const { container } = render(<Home />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
