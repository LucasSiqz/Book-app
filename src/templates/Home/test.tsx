import { render, screen, waitFor } from 'utils/test-utils'
import discoverMock from 'components/DiscoverCardSlider/mock'

import Home from '.'
import userEvent from '@testing-library/user-event'

jest.mock('components/WelcomeMessage', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock WelcomeMessage"></div>
    }
  }
})

jest.mock('components/Heading', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Heading"></div>
    }
  }
})

jest.mock('components/DiscoverCardSlider', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock DiscoverCardSlider"></div>
    }
  }
})

jest.mock('components/CurrentlyReading', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock CurrentlyReading"></div>
    }
  }
})

jest.mock('components/Review', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Review"></div>
    }
  }
})

jest.mock('components/BottomNavBar', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock BottomNavBar"></div>
    }
  }
})

jest.mock('components/Book', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Book"></div>
    }
  }
})

jest.mock('components/Loader', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Loader"></div>
    }
  }
})

const props = {
  discovery: {
    items: discoverMock
  },
  currentlyReading: {
    id: 'eLRhDgAAQBAJ',
    title: 'Originals',
    image: 'img/BookCover.png',
    author: 'Adam Grant',
    currentChapter: 2,
    chapterCount: 9
  },
  review: {
    title: "Don't Make Me Think",
    author: 'Jesse Showalter',
    image: 'img/Review.png',
    description: '<p>description</p>',
    viewsCount: 5.2,
    weeksAgo: 1
  }
}

describe('<Home />', () => {
  it('should render correctly', () => {
    render(<Home {...props} />)

    expect(screen.getByRole('textbox')).toBeInTheDocument()
    expect(screen.getByTestId('Mock WelcomeMessage')).toBeInTheDocument()
    expect(screen.getByTestId('Mock DiscoverCardSlider')).toBeInTheDocument()
    expect(screen.getByTestId('Mock CurrentlyReading')).toBeInTheDocument()
    expect(screen.getByTestId('Mock CurrentlyReading')).toBeInTheDocument()
    expect(screen.getByTestId('Mock Review')).toBeInTheDocument()
    expect(screen.getAllByTestId('Mock Heading')).toHaveLength(3)
  })

  it('should search books', async () => {
    render(<Home {...props} />)
    const input = screen.getByRole('textbox')
    const text = 'This is my book'
    userEvent.type(input, text)

    await waitFor(() => {
      expect(input).toHaveValue(text)
      expect(screen.getByTestId('Mock Loader')).toBeInTheDocument()
    })

    await waitFor(
      () => {
        expect(screen.getAllByTestId('Mock Book')).toHaveLength(9)
      },
      { timeout: 1500 }
    )
  })

  it('should load more books', async () => {
    render(<Home {...props} />)
    const input = screen.getByRole('textbox')
    const text = 'This is my book'
    userEvent.type(input, text)

    await waitFor(
      async () => {
        userEvent.click(screen.getByRole('button', { name: /Show More/i }))
        expect(screen.getByTestId('Mock Loader')).toBeInTheDocument()
      },
      { timeout: 1500 }
    )
  })
})
