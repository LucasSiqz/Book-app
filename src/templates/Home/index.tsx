import { useState } from 'react'

import discoverMock from 'components/DiscoverCardSlider/mock'
import TextField from 'components/TextField'
import WelcomeMessage from 'components/WelcomeMessage'
import Heading from 'components/Heading'
import DiscoverCardSlider from 'components/DiscoverCardSlider'
import CurrentlyReading from 'components/CurrentlyReading'
import Review from 'components/Review'
import BottomNavBar from 'components/BottomNavBar'
import { SearchIcon } from 'components/Icons'

import * as S from './styles'

const Home = () => {
  const [filter, setFilter] = useState('')

  return (
    <S.Wrapper>
      <S.Container>
        <S.HeaderSection>
          <TextField
            name="book"
            placeholder="Search book"
            icon={<SearchIcon />}
            value={filter}
            onChange={(event) => setFilter(event?.target.value)}
          />

          <WelcomeMessage name="Mehmed Al Fatih" />
        </S.HeaderSection>

        <S.DiscoverSection>
          <Heading
            title="Discover new book"
            nav={{
              title: 'More',
              path: '/books'
            }}
          />
          <DiscoverCardSlider items={discoverMock} />
        </S.DiscoverSection>

        <S.CurrentlyReadingSection>
          <Heading
            title="Currently Reading"
            nav={{
              title: 'All',
              path: '/book/originals'
            }}
          />
          <CurrentlyReading
            title="Originals"
            author="Adam Grant"
            currentChapter={2}
            chapterCount={9}
          />
        </S.CurrentlyReadingSection>

        <S.ReviewsSection>
          <Heading
            title="Reviews of The Days"
            nav={{
              title: 'All Video',
              path: '/book/originals'
            }}
          />
          <Review
            title="Don't Make Me Think"
            author="Jesse Showalter"
            description='"Dont Make Me Think" by Steve Krug is one of the first books I read when I was getting into digital design. It helped me move from designing things that just "look nice" to designing things that are usable, useful, memorable and simple to learn and use. '
            viewsCount={5.2}
            weeksAgo={1}
          />
        </S.ReviewsSection>
      </S.Container>

      <BottomNavBar />
    </S.Wrapper>
  )
}
export default Home
