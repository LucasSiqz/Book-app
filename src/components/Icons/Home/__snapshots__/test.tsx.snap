// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeIcon /> should render correctly 1`] = `
<svg
  fill="none"
  height="19"
  viewBox="0 0 18 19"
  width="18"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    clip-rule="evenodd"
    d="M1 6.22222L9 0L17 6.22222V16C17 16.9818 16.2041 17.7778 15.2222 17.7778H2.77778C1.79594 17.7778 1 16.9818 1 16V6.22222Z"
    fill-rule="evenodd"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
  />
  <path
    d="M6.33333 17.7778V8.88892H11.6667V17.7778"
    stroke="black"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
  />
</svg>
`
